editoast/src/core/mq_client.rs

```diff
@@ -0,0 +1,236 @@
+use futures_util::StreamExt;
+use lapin::{
+    options::{BasicPublishOptions, QueueDeclareOptions},
+    types::FieldTable,
+    BasicProperties, Channel, Connection, ConnectionProperties,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::to_vec;
+use std::fmt::Debug;
+use tokio::time::{timeout, Duration};
+use uuid::Uuid;
+
+pub struct RabbitMQClient {
+    connection: Connection,
+    core_queue_prefix: String,
+    exchange: String,
+    timeout: u64,
+}
+
+pub struct Options {
+    /// format `amqp://username:password@host:port/vhost`
+    /// for instance: `amqp://osrd:password@localhost:5672/%2f` for the default vhost
+    pub uri: String,
+    /// Exchange name
+    pub exchange: String,
+    /// Prefix for the core queues
+    pub core_queue_prefix: String,
+    /// Default timeout for the response
+    pub timeout: u64,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct MessageEnvelope<T>
+where
+    T: Serialize + Debug,
+{
+    payload: T,
+    message_type: String,
+    respond_to: Option<String>,
```

Comment:
I'd rather move the respond_to to amqp headers. it has several upsides:
- the worker can still respond even if it cannot deserialize the message
- if `message_type` and `infra_expected_version` are moved out of this layer, which I don't see any downside to, RPC messages are just `[u8]`. the business of message format and parsing can be moved out of this layer